http2_general/test_h2_block_action.py

```diff
@@ -431,6 +435,84 @@ def test_block_action_error_reply_with_conn_close(self, name, frame, expected_re
             self.check_no_fin_no_rst_in_sniffer(sniffer)
 
 
+class BlockActionH2ReplyFramesAfterShutdownWithCustomErrorPageSmallWindowSeveralInOneSkbBase(
+    BlockActionH2ReplyFramesAfterShutdownWithCustomErrorPageSmallWindowBase
+):
+    def _test(self, increment, count, invalid_wnd_update_frame):
+        client = self.get_client("deproxy")
+
+        sniffer = self.setup_sniffer()
+        self.start_services_and_initiate_conn(client)
+        if not invalid_wnd_update_frame:
+            self.save_must_fin_socks([client])
+            self.save_must_not_reset_socks([client])
+        else:
+            self.save_must_fin_socks([client])
+            self.save_must_reset_socks([client])
+        client.auto_flow_control = False
+
+        stream = client.init_stream_for_send(client.stream_id)
+        client.h2_connection.state_machine.process_input(ConnectionInputs.SEND_HEADERS)
+        stream_id = client.stream_id
+
+        headers_frame = HeadersFrame(
+            stream_id=stream_id,
+            data=HuffmanEncoder().encode(
+                [
+                    (":authority", "good.com"),
+                    (":path", "/"),
+                    (":scheme", "https"),
+                    (":method", "GET"),
+                    ("Content-Type", "invalid"),
+                ]
+            ),
+            flags=["END_HEADERS", "END_STREAM"],
+        )
+        invalid_frame = None
+        if invalid_wnd_update_frame:
+            invalid_frame = WindowUpdateFrame(0)
+            invalid_frame.window_increment = 0
+
+        conn_wnd_update_frame = WindowUpdateFrame(0)
+        conn_wnd_update_frame.window_increment = increment
+
+        client.h2_connection.state_machine.process_input(ConnectionInputs.SEND_WINDOW_UPDATE)
+        stream = client.h2_connection.streams[stream_id]
+        stream_wnd_update_frame = b""
+        for _ in range(0, count):
+            stream_wnd_update_frame += stream.increase_flow_control_window(increment)[0].serialize()
+
+        data = headers_frame.serialize()
+        if invalid_frame:
+            data += invalid_frame.serialize()
+        data += conn_wnd_update_frame.serialize() * count
+        data += stream_wnd_update_frame
+
+        client.send_bytes(data, expect_response=True)
+        self.assertTrue(client.wait_for_connection_close())
+        if not invalid_wnd_update_frame:
+            self.assertIsNotNone(client.last_response)
+            self.assertEqual(client.last_response.status, "400")
+            self.assertEqual(client.last_response.body, self.ERROR_RESPONSE_BODY)
+            self.check_fin_no_rst_in_sniffer(sniffer)
+        else:
+            self.assertFalse(client.last_response)
+            self.check_fin_and_rst_in_sniffer(sniffer)
+
+
+class BlockActionH2ReplyFramesAfterShutdownWithCustomErrorPageSmallWindowSeveralFramesInOneSkb(
```

Comment:
Can't we use the `parametrize_class` here? For example:
`{"name": "...", "ERROR_RESPONSE_BODY": "a" * 200, "INITIAL_WINDOW_SIZE": 100, "invalid_wnd_update_frame": True}`

Reply:
I parametrize method not class, because all this file is based on class inheritance. Also I can't move this class to `test_h2_frame` as you suggest, because I need to have one of the class from this file as a base class for the new tests